crates/sui-framework/src/natives/crypto.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use move_binary_format::errors::PartialVMResult;
+use move_vm_runtime::native_functions::NativeContext;
+use move_vm_types::{
+    gas_schedule::NativeCostIndex,
+    loaded_data::runtime_types::Type,
+    natives::function::{native_gas, NativeResult},
+    pop_arg,
+    values::Value,
+};
+use narwhal_crypto::traits::ToFromBytes;
+use smallvec::smallvec;
+use std::collections::VecDeque;
+
+pub fn ecrecover(
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 2);
+
+    let hashed_msg = pop_arg!(args, Vec<u8>);
+    let signature = pop_arg!(args, Vec<u8>);
+    let cost = native_gas(context.cost_table(), NativeCostIndex::EMIT_EVENT, 0);
```

Comment:
@oxade any advice on determining what gas cost should be?

Reply:
discussed offline, leaving as todo here with dummy cost, more accurate cost to follow up on https://github.com/MystenLabs/sui/issues/3593

Reply:
Yes we can skip cost discussions for now, we have to reiterate cost functions for many similar apis anyway. This PR shouldn't deal with cost estimation.